### ISSUE TITLE
Differentiation TKG/TCE from TMC needed in  Framework v0.11.1

### ISSUE THREAD

**Issue by AnilSankar75** [vmware-tanzu/community-edition#3319]
While we are working on the TCE/TKG pin mode from TMC side, we found there is no field of TCE for TMC to tell whether it is TKG or TCE. This would be a blocker for TMC STARTER in march as currently the plan is to pin this to  Framework v0.11.1. 
Because without the differentiation, it will be an issue from support perspective that the SRE/GSS/Engineer team to provide different support for TKG and TCE. It will introduce confusion and support effort for engineering team. 
Without the ability to differentiate TCE from TKG then in the TMC API/CLI/UI a TCE set-up will display as a TKG mgmt/workload cluster. UX is bad here. 

Additional reference : https://github.com/vmware-tanzu/tanzu-framework/pull/1599
## Environment Details

* Build version (`tanzu version`): 
* Deployment (Managed/Unmanaged cluster): 
* Infrastructure Provider (Docker/AWS/Azure/vSphere): 
* Operating System (client):

**Comment by joshrosso** [comment]
As stated in issue, code is complete in `main`. It should be backported in framework and validated in TCE before closing this issue.

@navidshaikh owns the backport; @joshrosso owns validation.

**Comment by joshrosso** [comment]
I am actively validating this against a new AWS cluster using:

* https://github.com/vmware-tanzu/community-edition/issues/3319

**Comment by joshrosso** [comment]
Edition detection is not working with the backported change.

```
$ k get cluster -A -o yaml | grep -A5 -i annotations
    annotations:
      TKGOperationInfo: '{"Operation":"Create","OperationStartTimestamp":"2022-03-07
        17:19:36.603454215 +0000 UTC","OperationTimeout":1800}'
      TKGOperationLastObservedTimestamp: 2022-03-07 17:25:06.65754813 +0000 UTC
      TKGVERSION: v1.5.0-tf-v0.11.1
      kubectl.kubernetes.io/last-applied-configuration: |
        {"apiVersion":"cluster.x-k8s.io/v1beta1","kind":"Cluster","metadata":{"annotations":{"osInfo":"ubuntu,20.04,amd64","tkg/plan":"dev"},"labels":{"cluster-role.tkg.tanzu.vmware.com/management":"","tanzuKubernetesRelease":"v1.22.3---vmware.1-tkg.1-tf-v0.11.1","tkg.tanzu.vmware.com/cluster-name":"newit"},"name":"newit","namespace":"tkg-system"},"spec":{"clusterNetwork":{"pods":{"cidrBlocks":["100.96.0.0/11"]},"services":{"cidrBlocks":["100.64.0.0/13"]}},"controlPlaneRef":{"apiVersion":"controlplane.cluster.x-k8s.io/v1beta1","kind":"KubeadmControlPlane","name":"newit-control-plane"},"infrastructureRef":{"apiVersion":"infrastructure.cluster.x-k8s.io/v1beta1","kind":"AWSCluster","name":"newit"}}}
      osInfo: ubuntu,20.04,amd64
      tkg/plan: dev
    creationTimestamp: "2022-03-07T17:30:55Z"
    finalizers:
    - cluster.cluster.x-k8s.io
```

I'm digging in now to see if this backported patch is making assumption about things that were perhaps only available in v0.17.0+.

cc @navidshaikh s as this could mean more backporting required.

**Comment by joshrosso** [comment]
Yup, that was my fault.

Edition is now exposed:

```
$ k get cluster -A -o yaml | grep -A5 -i annotations
    annotations:
      TKGOperationInfo: '{"Operation":"Create","OperationStartTimestamp":"2022-03-07
        19:07:35.017942559 +0000 UTC","OperationTimeout":1800}'
      TKGOperationLastObservedTimestamp: 2022-03-07 19:12:50.065037833 +0000 UTC
      TKGVERSION: v1.5.0-tf-v0.11.0
      edition: tce
--
        {"apiVersion":"cluster.x-k8s.io/v1beta1","kind":"Cluster","metadata":{"annotations":{"osInfo":"amazon,2,amd64","tkg/plan":"dev"},"labels":{"cluster-role.tkg.tanzu.vmware.com/management":"","tanzuKubernetesRelease":"v1.22.3---vmware.1-tkg.1-tf-v0.11.0","tkg.tanzu.vmware.com/cluster-name":"test"},"name":"test","namespace":"tkg-system"},"spec":{"clusterNetwork":{"pods":{"cidrBlocks":["100.96.0.0/11"]},"services":{"cidrBlocks":["100.64.0.0/13"]}},"controlPlaneRef":{"apiVersion":"controlplane.cluster.x-k8s.io/v1beta1","kind":"KubeadmControlPlane","name":"test-control-plane"},"infrastructureRef":{"apiVersion":"infrastructure.cluster.x-k8s.io/v1beta1","kind":"AWSCluster","name":"test"}}}
      osInfo: amazon,2,amd64
      tkg/plan: dev
    creationTimestamp: "2022-03-07T19:18:39Z"
    finalizers:
    - cluster.cluster.x-k8s.io
```
